test/test_hessian.py

```diff
@@ -88,16 +46,12 @@ def test_blocked_HessianLinearOperator_matmat(
     )
 
     # compute the blocks with functorch and build the block diagonal matrix
-    op_functorch = [
-        functorch_hessian(
-            model_func, loss_func, params_block, data, input_key="x"
-        ).detach()
-        for params_block in split_list(params, block_sizes)
+    H_blocks = [
+        functorch_hessian(model_func, loss_func, params_block, data, input_key="x")
+        for params_block in split_list(
+            params, [len(params)] if block_sizes is None else block_sizes
+        )
     ]
-    op_functorch = block_diag(*op_functorch).cpu().numpy()
-
-    if adjoint:
-        op, op_functorch = op.adjoint(), op_functorch.conj().T
+    H_mat = block_diag(*H_blocks)
 
-    X = random.rand(op.shape[1], num_vecs)
-    report_nonclose(op @ X, op_functorch @ X, atol=1e-6, rtol=5e-4)
+    compare_matmat(H, H_mat, adjoint, is_vec, rtol=1e-4, atol=1e-6)
```

Comment:
One case of `block_sizes_fn` corresponds to the full Hessian.

Reply:
Ok, I was thinking about the case of an NN with multiple layers and the full Hessian. But this might make no actual difference to what the test covers, so I guess it's fine.